Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -245,8 +245,16 @@ private extension TabScrollingController {
     }
 
     func animateToolbarsWithOffsets(_ animated: Bool, duration: TimeInterval, headerOffset: CGFloat, footerOffset: CGFloat, alpha: CGFloat, completion: ((_ finished: Bool) -> Void)?) {
+        guard let scrollView = scrollView else { return }
+        let initialContentOffset = scrollView.contentOffset
+
+        // If this function is used to fully animate the toolbar from hidden to shown, keep the page from scrolling by adjusting contentOffset.
+        let isShowFromScratch = self.headerTopOffset == -UIConstants.TopToolbarHeight && headerOffset == 0
 
         let animation: () -> Void = {
+            if isShowFromScratch {
```

Comment:
I think this could be `isShownFromScratch` or `showingFromScratch`?
I also think `isShownFromHidden` or `showingFromHidden` works better in line with UIView terminology

Reply:
thanks, I was stumped for a good name for this